python/cudf/cudf/tests/test_concat.py

```diff
@@ -1981,19 +1981,23 @@ def test_concat_dictionary(d, axis):
 @pytest.mark.parametrize(
     "d",
     [
-        {"first": cudf.Index([1, 2, 3])},
+        {"first": (cudf.Index, {"data": [1, 2, 3]})},
         {
-            "first": cudf.MultiIndex(
-                levels=[[1, 2], ["blue", "red"]],
-                codes=[[0, 0, 1, 1], [1, 0, 1, 0]],
+            "first": (
+                cudf.MultiIndex,
+                {
+                    "levels": [[1, 2], ["blue", "red"]],
+                    "codes": [[0, 0, 1, 1], [1, 0, 1, 0]],
+                },
             )
         },
-        {"first": cudf.CategoricalIndex([1, 2, 3])},
+        {"first": (cudf.CategoricalIndex, {"data": [1, 2, 3]})},
     ],
 )
 def test_concat_dict_incorrect_type_index(d):
     with pytest.raises(
         TypeError,
         match="cannot concatenate a dictionary containing indices",
     ):
-        cudf.concat(d, axis=1)
+        _dict = {k: c(**v) for k, (c, v) in d.items()}
+        cudf.concat(_dict, axis=1)
```

Comment:
It's probably okay to overwrite the input variable name here.
```suggestion
        d = {k: c(**v) for k, (c, v) in d.items()}
        cudf.concat(d, axis=1)
```